stac_fastapi/types/stac_fastapi/types/hierarchy.py

```diff
@@ -0,0 +1,200 @@
+    conformance_classes: List[str],
+    extension_schemas: List[str],
+) -> stac_types.LandingPage:
+    """Generate a STAC API landing page/catalog."""
+    catalog_links = [
+        browseable_catalog_link(
+            child, base_url, "/".join([catalog_path.strip("/"), child["catalog_id"]])
+        )
+        for child in catalog_node["children"]
+        if "catalog_id" in child
+    ]
+    collection_links = [
+        browseable_collection_link(child, base_url)
+        for child in catalog_node["children"]
+        if "collection_id" in child
+    ]
+    children_links = catalog_links + collection_links
+    item_links = [
+        browseable_item_link(item, base_url) for item in catalog_node["items"]
+    ]
+
+    split_catalog_path = catalog_path.split("/")
+    if len(split_catalog_path) > 1:
+        parent_href = urljoin(
+            base_url, f"/catalogs/{'/'.join(split_catalog_path[:-1])}"
+        )
+    else:
+        parent_href = base_url
+
+    standard_links = [
+        {
+            "rel": Relations.self.value,
+            "type": MimeTypes.json,
+            "href": urljoin(base_url, f"/catalogs/{catalog_path.strip('/')}"),
+        },
+        {
+            "rel": Relations.root.value,
+            "type": MimeTypes.json,
+            "href": base_url,
+        },
+        {
+            "rel": "data",
+            "type": MimeTypes.json,
+            "href": urljoin(
+                base_url, f"/catalogs/{catalog_path.strip('/')}/collections"
+            ),
+        },
+        {
+            "rel": Relations.conformance.value,
+            "type": MimeTypes.json,
+            "title": "STAC/WFS3 conformance classes implemented by this api",
+            "href": urljoin(
+                base_url, f"/catalogs/{catalog_path.strip('/')}/conformance"
+            ),
+        },
+        {
+            "rel": Relations.search.value,
+            "type": MimeTypes.geojson,
+            "title": "STAC search",
+            "href": urljoin(base_url, f"/catalogs/{catalog_path.strip('/')}/search"),
+            "method": "GET",
+        },
+        {
+            "rel": Relations.search.value,
+            "type": MimeTypes.json,
+            "title": "STAC search",
+            "href": urljoin(base_url, f"/catalogs/{catalog_path.strip('/')}/search"),
+            "method": "POST",
+        },
+        {"rel": Relations.parent.value, "type": MimeTypes.json, "href": parent_href},
+    ]
+
+    catalog_page = stac_types.LandingPage(
+        type="Catalog",
+        id=catalog_node["catalog_id"],
+        title=catalog_node["catalog_id"],
+        description=catalog_node["description"],
+        stac_version=stac_version,
+        conformsTo=conformance_classes,
+        links=standard_links + children_links + item_links,
+        stac_extensions=extension_schemas,
+    )
+    return catalog_page
+
+
+# PGSTAC TODO: function to allow ingest of hierarchy
+def parse_hierarchy(d: dict) -> BrowseableNode:
```

Comment:
Return type should be a Union of all three nodes, but this function is only ever called on a dictionary that would return a `BrowseableNode`.  So I'm not sure if the return type is wrong or the `CollectionNode` and `CatalogNode` returns aren't needed.

Reply:
It is very possible that I'm missing something peculiar about python types or `TypedDict` inheritance, but I think we can avoid the `Union` here because of the inheritance structure - both `CollectionNode` and `CatalogNode` are subsumed under the `BrowseableNode` type. Let me know if I'm off base in this assertion